atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTableTest.java

```diff
@@ -347,6 +348,104 @@ public void createDoesNotThrowWhenOverflowColumnAlreadyExistsAndAlterIsRan() thr
         verifyTableAltered();
     }
 
+    @Test
+    public void createThrowsIfAlterConfiguredAndMetadataNeedsUpdateAndMigrationInProgress() {
+        assertThatThrownBy(() -> testAlterMetadataForMigrationState(OverflowMigrationState.IN_PROGRESS))
+                .isInstanceOf(SafeIllegalArgumentException.class)
+                .hasMessageContaining(MISSING_OVERFLOW_EXCEPTION_MESSAGE);
+        verifyTableSizeMetadataNotUpdated();
+    }
+
+    @Test
+    public void createThrowsIfAlterConfiguredAndMetadataNeedsUpdateAndMigrationUnstarted() {
+        assertThatThrownBy(() -> testAlterMetadataForMigrationState(OverflowMigrationState.UNSTARTED))
+                .isInstanceOf(SafeIllegalArgumentException.class)
+                .hasMessageContaining(MISSING_OVERFLOW_EXCEPTION_MESSAGE);
+        verifyTableSizeMetadataNotUpdated();
+    }
+
+    @Test
+    public void createThrowsIfAlterConfiguredAndMetadataNeedsUpdateAndMigrationFinishing() {
+        assertThatThrownBy(() -> testAlterMetadataForMigrationState(OverflowMigrationState.FINISHING))
```

Comment:
IMO this is why I typically opt for testing the helper method here, as this leads to large unit test bloat